drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c

```diff
@@ -205,9 +204,7 @@ void nrf_wifi_scan_timeout_work(struct k_work *work)
 
 #ifdef CONFIG_NET_L2_WIFI_MGMT
 	if (disp_scan_cb) {
-		memset(&res, 0x0, sizeof(res));
-
-		disp_scan_cb(vif_ctx_zep->zep_net_if_ctx, -ETIMEDOUT, &res);
+		disp_scan_cb(vif_ctx_zep->zep_net_if_ctx, -ETIMEDOUT, NULL);
```

Comment:
`In case of scan timeout, driver should indicate scan done with status as set to timed out, but instead an empty scan result is being sent.`

Reply:
Yes, the fix here is sending `NULL` instead of empty result which means scan is done.

Reply:
Reworded the commit message.

Reply:
> The change here seems to be removal of the structure that was being declared. The status was still being set to -ETIMEDOUT Right?

There is no change in status, yes removed intermediate structs.